### PR TITLE
Initialize attached disk in windows ARM

### PR DESCRIPTION
**Pull request by nakomis**
This supercedes #35

### CODE REVIEW

File: blockstore/src/main/java/brooklyn/location/blockstore/AbstractVolumeManager.java [hunk 1]
```diff
@@ -73,58 +75,67 @@ public MountedBlockDevice attachAndMountVolume(JcloudsMachineLocation machine, B
     @Override
     public void createFilesystem(AttachedBlockDevice attachedDevice, FilesystemOptions filesystemOptions) {
         JcloudsMachineLocation machine = attachedDevice.getMachine();
-        if (!(machine instanceof SshMachineLocation)) {
-            throw new IllegalStateException("Cannot create filesystem for "+machine+" of type "+machine.getClass().getName()+"; expected "+SshMachineLocation.class.getSimpleName());
```
Comment (aledsage) [inline blockstore/src/main/java/brooklyn/location/blockstore/AbstractVolumeManager.java]: Should we still throw this in an else block (if it's not `SshMacheLocation` or `WinRmMachineLocation`)?

File: blockstore/src/main/java/brooklyn/location/blockstore/AbstractVolumeManager.java [hunk 2]
```diff
@@ -73,58 +75,67 @@ public MountedBlockDevice attachAndMountVolume(JcloudsMachineLocation machine, B
...
+                        machine, filesystemOptions.getFilesystemType(), response.getStdErr()));
+            }
         }
     }
 
     @Override
     public MountedBlockDevice mountFilesystem(AttachedBlockDevice attachedDevice, FilesystemOptions options) {
         JcloudsMachineLocation machine = attachedDevice.getMachine();
-        if (!(machine instanceof SshMachineLocation)) {
-            throw new IllegalStateException("Cannot mount filesystem for "+machine+" of type "+machine.getClass().getName()+"; expected "+SshMachineLocation.class.getSimpleName());
```
Comment (aledsage) [inline blockstore/src/main/java/brooklyn/location/blockstore/AbstractVolumeManager.java]: Again, should we fail if it's not ssh or winrm?

**Comment by nakomis** [comment]
@aledsage I've made the changes you requested, and am happy to look at automated tests (I think the `catalog.tests.bom` approach is best as that will actually get run), but can this PR be merged beforehand so that it makes it into the release?

**Comment by aledsage** [comment]
@nakomis can you try to grab someone to test it on windows, or say what test you ran (e.g. which cloud/image/configuration, and example yaml for doing that)?

If it's not merged by Monday lunchtime then please do pester me again though!

**Comment by nakomis** [comment]
@aledsage Sure. Here's the test YAML I used:

```
name: Windows with M Drive
location: west-win-azure
services:
- type: org.apache.brooklyn.entity.software.base.VanillaWindowsProcess
  name: Windows 2012 R2 Datacenter
  brooklyn.config:
    checkRunning.command: echo true
    customize.command: echo true
    lanch.command: echo true
    provisioning.properties:
      userMetadata:
        keep: keep
      customizers:
      - $brooklyn:object:
          type: io.brooklyn.blockstore.brooklyn-blockstore:brooklyn.location.blockstore.NewVolumeCustomizer
          object.fields:
            volumes:
            - blockDevice:
                sizeInGb: 3
                deviceSuffix: 'h'
                deleteOnTermination: true
              filesystem:
                mountPoint: M
                filesystemType: NTFS
                volumeLabel: My M Drive
```

And here's the location definition I used (creds obfuscated):

```
brooklyn.catalog:
  id: west-win-azure
  name: west-win-azure
  itemType: location
  item:
    type: jclouds:azurecompute-arm
    brooklyn.config:
      identity: <removed>
      credential: <removed>
      endpoint: https://management.azure.com/subscriptions/<removed>
      oauth.endpoint: https://login.microsoftonline.com/<removed>/oauth2/token
      jclouds.azurecompute.arm.publishers: MicrosoftWindowsServer
      jclouds.azurecompute.operation.timeout: 120000
 
      winrm.useHttps: true
      osFamily: windows
      imageId: westeurope/MicrosoftWindowsServer/WindowsServer/2012-R2-Datacenter
      region: westeurope
      vmNameMaxLength: 15
      useJcloudsSshInit: false
      destroyOnFailure: false
 
      templateOptions:
        overrideLoginUser: cloudsoft
        overrideLoginPassword: "<removed>"
        secrets:
        - sourceVault:
            id: "<removed>"
          vaultCertificates:
          - certificateUrl: "<removed>"
            certificateStore: My
        windowsConfiguration:
          provisionVMAgent: true
          winRM:
            listeners:
            - protocol: https
              certificateUrl: "<removed>"
          additionalUnattendContent: null
          enableAutomaticUpdates: true
```

I'll see if I can find someone to test it...

**Comment by aledsage** [comment]
Thanks @nakomis. I believe that @rdowner has been looking at this with you to test and review it. He's happy with how you've tested it, but is still getting his cloud account / config for repeating your tests. As such, I'm happy to merge now.

**Pull request merged**